cli/genkey/genkey_test.go

```diff
@@ -1 +1,42 @@
 package genkey
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/cloudflare/cfssl/cli"
+	"os"
+	"os/exec"
+	"path"
+	"testing"
+)
+
+func TestGenkey(t *testing.T) {
+	gopath := os.Getenv("GOPATH")
+	cfssl := path.Join(gopath, "bin", "cfssl")
+	testdata := path.Join(gopath, "src", "github.com", "cloudflare", "cfssl", "testdata")
+
+	out1, err1 := exec.Command(cfssl, "genkey", path.Join(testdata, "csr.json")).Output()
+	var response map[string]interface{}
+	json.Unmarshal(out1, &response)
+	fmt.Println("Error: ", err1)
+	fmt.Println(response["key"])
+	fmt.Println(response["csr"])
+
+	if err1 != nil {
+		t.Fatal(err1)
+	}
+
+	c := cli.Config{}
+
+	err2 := genkeyMain([]string{path.Join(testdata, "csr.json")}, c)
+	if err2 != nil {
+		t.Fatal(err2)
+	}
+
+	c.IsCA = true
+
+	err3 := genkeyMain([]string{path.Join(testdata, "csr.json")}, c)
+	if err2 != nil {
+		t.Fatal(err3)
```

Comment:
just reuse 'err'. As you can see here, it is problematic and error-prone to use adhoc new variables and copy/paste old logic.